packages/stream-metadata/src/routes/media.ts

```diff
@@ -53,8 +56,19 @@ export async function fetchMedia(request: FastifyRequest, reply: FastifyReply) {
 	}
 
 	const { mediaStreamId } = paramsResult.data
-	const { key, iv } = queryResult.data
-	logger.info({ mediaStreamId, key, iv }, 'Fetching media stream content')
+	const { key, iv, size } = queryResult.data
+
+	// Parse and validate size parameter
+	const sizeOptions = parseSizeParam(size)
+	if (!sizeOptions.isValid) {
+		logger.info({ size }, 'Invalid size parameter format')
+		return reply
+			.code(400)
+			.header('Cache-Control', CACHE_CONTROL['4xx'])
+			.send({ error: 'Bad Request', message: 'Size must be in format WxH, xH, or Wx' })
+	}
```

Comment:
do we require a size now?

Reply:
an `undefined` size queryParam is a valid size queryParam  🤷 

```ts
export function parseSizeParam(size?: string): ParsedSizeParam {
	if (!size) {
		return { isValid: true } // No size param is valid (original size)
	}
    ...
```